src/utils/status.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { Tooltip, Icon } from 'antd'
+
+const isAutomaticallyUpgradeEngine = (volume, engineUpgradePerNodeLimit, defaultEngineImage) => {
+  if (engineUpgradePerNodeLimit && engineUpgradePerNodeLimit.value !== '0') {
+    if (defaultEngineImage) {
+      return volume.engineImage !== defaultEngineImage.image
+    }
+    return false
+  }
+  return false
+}
+
+
+export function statusUpgradingEngine(volume, engineUpgradePerNodeLimit, engineImages) {
+  let oldSize = volume && volume.controllers && volume.controllers[0] && volume.controllers[0].size ? volume.controllers[0].size : ''
+  let defaultEngineImage = engineImages.find(engineImage => engineImage.default)
+  // The condition for determining that an engine is being upgraded
+  // - Not DR volume
+  // - Not degraded volume
+  // - Not expanding volume
+  // - In upgrading process (i.g., curretnImage != engineImage )
+  // - concurrent-automatic-engine-upgrade-per-node-limit setting is bigger than 0
+  if (!volume.standby && volume.robustness !== 'degraded' && volume.size === oldSize && isAutomaticallyUpgradeEngine(volume, engineUpgradePerNodeLimit, defaultEngineImage)) {
+    return (<Tooltip title={`Volume engine is being upgraded from ${volume.engineImage} to ${defaultEngineImage.image}`}><Icon style={{ marginRight: 5 }} type="sync" spin /></Tooltip>)
+  }
+  return ''
+}
```

Comment:
This is the condition to show the global notification. We don't need this to check if an individual volume is upgrading.